ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java

```diff
@@ -46,7 +46,7 @@
 
   private Expression havingExpression = null;
 
-  private Optional<Integer> limitClause = Optional.empty();
```

Comment:
why would you leave Optional? Why not use functional style and avoid dealing with NPE?

Reply:
It won't be null as users of this class will call `getLimitClause()` that returns `Optional.ofNullable(limitClause)` - there is no reason for it to be `Optional` within the class